src/storage/buffer/buffer_pool.cpp

```diff
@@ -1,12 +1,41 @@
 #include "duckdb/storage/buffer/buffer_pool.hpp"
 
+#include "duckdb/common/chrono.hpp"
 #include "duckdb/common/exception.hpp"
 #include "duckdb/parallel/concurrentqueue.hpp"
 #include "duckdb/storage/temporary_memory_manager.hpp"
-#include "duckdb/common/chrono.hpp"
+#include <cstddef>
+#include <cstdint>
+
+#if defined(_WIN32)
+#include <windows.h>
+#elif defined(__GNUC__)
+#include <sched.h>
+#include <unistd.h>
+#endif
 
 namespace duckdb {
 
+// this code comes from jemalloc
+static int GetCurrentCPU() {
+#if defined(_WIN32)
+	return (int)GetCurrentProcessorNumber();
+#elif defined(DUCKDB_HAVE_SCHED_GETCPU)
+	return (int)sched_getcpu();
+#elif defined(DUCKDB_HAVE_RDTSCP)
+	unsigned int ecx;
+	asm volatile("rdtscp" : "=c"(ecx)::"eax", "edx");
+	return (int)(ecx & 0xfff);
+#elif defined(__aarch64__) && defined(__APPLE__)
+	/* Other oses most likely use tpidr_el0 instead */
+	uintptr_t c;
+	asm volatile("mrs %x0, tpidrro_el0" : "=r"(c)::"memory");
+	return (int)(c & (1 << 3) - 1);
+#else
+	return -1;
```

Comment:
Maybe we can use `std::this_thread::get_id` here, either entirely as a replacement for CPU ID, or at least as a fallback in case we cannot obtain a valid CPU ID?